cpp/src_prims/random/make_blobs.cuh

```diff
@@ -26,46 +28,120 @@
 namespace MLCommon {
 namespace Random {
 
+namespace {
+
+// generate the labels first and shuffle them instead of shuffling the dataset
+template <typename IdxT>
+void generate_labels(IdxT* labels, IdxT n_rows, IdxT n_clusters, bool shuffle,
+                     cudaStream_t stream) {
+  // always keep 'a' to be coprime to n_rows
+  IdxT a = rand() % n_rows;
+  while (gcd(a, n_rows) != 1) a = (a + 1) % n_rows;
+  IdxT b = rand() % n_rows;
+  auto op = [=] __device__(IdxT * ptr, IdxT idx) {
+    if (shuffle) {
+      idx = IdxT((a * int64_t(idx)) + b) % n_rows;
+    }
+    *ptr = idx % n_clusters;
+  };
+  LinAlg::writeOnlyUnaryOp<IdxT, decltype(op), IdxT>(labels, n_rows, op,
+                                                     stream);
+}
+
+template <typename DataT, typename IdxT>
+DI void get_mu_sigma(DataT& mu, DataT& sigma, IdxT idx, const IdxT* labels,
```

Comment:
I revamped whole of the existing make_blobs unit-tests to:
1. generate uniform random centers apriori to make_blobs call
2. recompute the label-wise mean and variance of the generated dataset
3. compare the above random centers to this computed mean

I believe this should test both `get_mu_sigma` as well as `generate_labels` correctly. Because if any one is not being done correctly, the final computed mean/variance will be incorrect, thereby failing the unit-tests. What do you think about the updated unit-tests now?